src/analysis.cc

```diff
@@ -392,4 +392,56 @@ bool analysis_t::erase(const ip_prefix_t &ip_prefix, source_t source,
   return status;
 }
 
+timestamps_t intersect(const timestamps_t &a, const timestamps_t &b) {
```

Comment:
Not to be pedantic, but it is confusing to mix already applied and yet-to-be-applied changes. I mainly asking, because I also want to improve the quality of patches so it's easier to track changes. Is it possible to remove the C++ changes?